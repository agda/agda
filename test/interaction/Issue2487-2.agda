{-# OPTIONS --safe #-}
module Issue2487-2 where

import Issue2487.A
