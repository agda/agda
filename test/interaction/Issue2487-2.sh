AGDA_BIN=$1

# A works with --safe
$AGDA_BIN --safe Issue2487/A.agda

# Works with --safe, imports A without rechecking
$AGDA_BIN --safe -v import.iface.options:4 Issue2487-2.agda

# Doesn't work without --safe, because A is rechecked (options changed)
$AGDA_BIN -v import.iface.options:4 Issue2487-2.agda
