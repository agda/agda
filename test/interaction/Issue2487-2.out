Checking Issue2487.A (Issue2487/A.agda).
Checking Issue2487-2 (Issue2487-2.agda).
Changes in the following options, re-typechecking: [--no-sized-types, --no-guardedness, --safe]
Checking Issue2487.A (Issue2487/A.agda).
Importing module Issue2487.A not using the --safe flag from a
module which does.
Importing module Issue2487.A not using the --no-sized-types flag
from a module which does.
Importing module Issue2487.A not using the --no-guardedness flag
from a module which does.
